cloudinit/net/__init__.py

```diff
@@ -1034,6 +1034,16 @@ def get_interfaces_by_mac_on_linux(blacklist_drivers=None) -> dict:
                         % (ret[mac], driver_map[mac], name)
                     )
 
+            # This is intended to be a short-term fix of LP: #1997922
+            # Long term, we should better handle configuration of virtual
+            # devices where duplicate MACs are expected early in boot if
+            # cloud-init happens to enumerate network interfaces before drivers
+            # have fully initialized the leader/subordinate relationships for
+            # those devices or switches.
+            if driver == "mscc_felix" or driver == "fsl_enetc":
+                raise_duplicate_mac_error = False
+                continue
```

Comment:
Since we are issuing`continue` in this case we actually don't need to set `raise_duplicate_mac_error` as we won't consume that anywhere. How about if we just log a similar warning breadcrumb that we do in Azure. As this is currently written, we silently ignore the duplicate macs and don't log any warnings.

```suggestion
+            if driver in ("mscc_felix", "fsl_enetc"):
                LOG.warning(
                    msg + " Ignoring '%s' due to driver '%s' and "
                    "'%s' having driver '%s'."
                    % (name, driver, ret[mac], driver_map[mac])
                )
                continue
```